File: Recursion/SumOfDigit/sumOfDigit.py
def sumOfDigits(num):

    assert num>=0 and int(num) == num , "The Number has to be a positive integer"

    if num == 0:
        return num

    else:
        return num%10 + sumOfDigits(num//10)


user_input = int(input("Enter number: "))
print(sumOfDigits(user_input))

File: Recursion/SumOfDigit/sumOfDigit.js
//Main Method
function sumOfDigits(num){
        if(num==0)
        {
            return num;
        }
        else
        {
            return (num%10) + sumOfDigits(num/10);
        }
}


const ps = require("prompt-sync");
const prompt = ps();

//User Input
//Console Method
const user = ()=>{
    let user;
    while(true)
    {
        user = parseInt(prompt("Enter Number: "));
        if(user>=0)
        {
            break;
        }
        console.log("The number has to be a positive integer only");

    }
    return user;

}

const inp = user();

const result = Math.floor(sumOfDigits(inp));
console.log(result);